ElementX/Sources/Screens/JoinRoomScreen/View/JoinRoomScreen.swift

```diff
@@ -66,18 +76,69 @@ struct JoinRoomScreen: View {
                         .multilineTextAlignment(.center)
                         .lineLimit(3)
                 }
+                
+                if context.viewState.mode == .knock {
+                    Spacer()
+                        .frame(height: 19)
+                    knockMessage
+                }
+            }
+        }
+    }
+    
+    @ViewBuilder
+    private var knockedView: some View {
+        VStack(spacing: 16) {
+            HeroImage(icon: \.checkCircleSolid, style: .success)
+            VStack(spacing: 8) {
+                Text(L10n.screenJoinRoomKnockSentTitle)
+                    .font(.compound.headingMDBold)
+                    .foregroundStyle(.compound.textPrimary)
+                    .multilineTextAlignment(.center)
+                Text(L10n.screenJoinRoomKnockSentDescription)
+                    .font(.compound.bodyMD)
+                    .foregroundStyle(.compound.textSecondary)
+                    .multilineTextAlignment(.center)
             }
         }
     }
     
+    @ViewBuilder
+    private var knockMessage: some View {
+        VStack(alignment: .leading, spacing: 12) {
+            HStack(spacing: 0) {
+                TextField("", text: $context.knockMessage, axis: .vertical)
+                    .onChange(of: context.knockMessage) { newValue in
+                        context.knockMessage = String(newValue.prefix(1000))
+                    }
+                    .lineLimit(4, reservesSpace: true)
+                    .font(.compound.bodyMD)
+                    .padding(.horizontal, 16)
+                    .padding(.vertical, 12)
+            }
+            .background(.compound.bgCanvasDefault)
+            .cornerRadius(8)
+            .overlay(RoundedRectangle(cornerRadius: 8)
+                .inset(by: 0.5)
+                .stroke(.compound.borderInteractivePrimary))
```

Comment:
Technically I think that `overlay(some View)` is deprecated?
```suggestion
            .overlay {
                RoundedRectangle(cornerRadius: 8)
                    .inset(by: 0.5)
                    .stroke(.compound.borderInteractivePrimary)
            }
```